extensions/amp-story/1.0/media-pool.js

```diff
@@ -265,7 +270,7 @@ export class MediaPool {
             // Use seed element at end of set to prevent wasting it.
             (i == 1 ? mediaElSeed : mediaElSeed.cloneNode(/* deep */ true));
         const sources = this.getDefaultSource_(type);
-        mediaEl.setAttribute('pool-element', elId++);
+        mediaEl.id = POOL_ELEMENT_ID_PREFIX + poolIdCounter++;
```

Comment:
IDs have more constraints than attributes, I don't really mind, but is there a reason why it'd be easier to use IDs here?

Reply:
My two reasons were just:

1. I just felt that it was more semantic.  Looking at the DOM and seeing just `<video id="i-amphtml-pool-media-4">`, I feel like I know that this is a unique identifier for this video (and perhaps that this video is the nth media element created by the pool).  But seeing `<video pool-element="i-amphtml-pool-media-4">`, I'm not sure what that means.  Is that a different element that this video is related to?  Is that a setting of some sort?  I have to understand MediaPool's internal implementation to understand what this means.  Arguably, this is a naming thing; we could probably also just decide a better name for `pool-element`, but I just felt `id` already represented that.
2. The more important thing is that it matches what we were already doing for placeholder elements, so that at a glance, when debugging for like 10 hours, you can tell which elements are which.  Arguably, you could instead switch placeholder elements to use a well-named attribute, but I didn't for the reasons stated above.

I don't think the constraints on ID hurt us here, though